### PR TITLE
Add pattern regarding using SimpleDelegator instead of method_missing

### PR DESCRIPTION
**Pull request by bockets**

### CODE REVIEW

**Review by karlentwistle — APPROVED** [review]

File: _patterns/prefer-simple-delegator-over-method-missing.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
+---
+categories: Rails
+name: Prefer SimpleDelegator over method_missing for wrapper classes
+---
+
+If you find yourself needing make a light wrapper for an object to extend it with new methods, prefer using the standard
```
Comment (RajaaBoulassouak) [inline _patterns/prefer-simple-delegator-over-method-missing.md:6]:
```suggestion
If you find yourself needing to make a light wrapper for an object to extend it with new methods, prefer using the standard
```

**Review by RajaaBoulassouak — APPROVED** [review]

**Review by baerbradford — APPROVED** [review]

**Review by bhanubhakta — APPROVED** [review]

**Review by cernivec — APPROVED** [review]

**Comment by bockets** [comment]
Holding on nitpicks from @johngallagher

File: _patterns/prefer-simple-delegator-over-method-missing.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    wrapped.respond_to?(meth)
+  end
+end
+```
+
+## Good
+
+```ruby
+class Wrapper < SimpleDelegator
+  attr_reader :wrapped
```
Comment (johngallagher) [inline _patterns/prefer-simple-delegator-over-method-missing.md:43]: Could we move this to be a `private` `attr_reader`? Internals should be hidden by default unless there's a good reason to make them public. :)

```suggestion
  attr_reader :wrapped
```

File: _patterns/prefer-simple-delegator-over-method-missing.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+  def initialize(wrapped)
+    super(wrapped)
+    @wrapped = wrapped
+  end
+
+  def expired?
+    wrapped.fetch(:expired_at) > Time.now
```
Comment (johngallagher) [inline _patterns/prefer-simple-delegator-over-method-missing.md:51]: I know it's not the point of the example, but everything in Rails related to time should be `Time.zone`

```suggestion
    wrapped.fetch(:expired_at) > Time.zone.now
```

**Review by johngallagher — APPROVED** [review]
Approving - just a couple of tweaks needed but no hills I'm willing to die on. :)

File: _patterns/prefer-simple-delegator-over-method-missing.md (hunk 3 above)
Comment (bockets) [reply _patterns/prefer-simple-delegator-over-method-missing.md:51]: Hmm good point. I don't want this to be Rails-specific, but I put it in the Rails category 😅 so it kinda is.

File: _patterns/prefer-simple-delegator-over-method-missing.md (hunk 2 above)
Comment (bockets) [reply _patterns/prefer-simple-delegator-over-method-missing.md:43]: Interesting idea. Let me try that out in some code and get back to you on this.

**Pull request merged**